i8.go

```diff
@@ -12,7 +12,11 @@ func (value I8) Bytes() []byte {
 	return U8(value).Bytes()
 }
 
-func DecodeI8(buffer *bytes.Buffer) I8 {
+func DecodeI8(buffer *bytes.Buffer) (I8, error) {
 	decoder := Decoder{Reader: buffer}
-	return I8(decoder.DecodeByte())
+	db, err := decoder.DecodeByte()
```

Comment:
We should try to keep things consistent, in one place the return number is `dec64/dec32/...`, here is `db`.
Why not `dec8`?